repl-test-update/run-insert-error.scala

```diff
@@ -0,0 +1,52 @@
+def replTestDir = "../compiler/test-resources"
+
+def list(dir: String = "."): Vector[String] =
+    Option(java.io.File(dir).list).map(_.toVector).getOrElse(Vector())
+
+def loadLines(fileName: String, enc: String = "UTF-8"): Vector[String] = 
+  var result = Vector.empty[String]
+  val source = scala.io.Source.fromFile(fileName, enc)
+  try result = source.getLines().toVector finally source.close()
+  result
+
+def saveString(text: String, fileName: String, enc: String = "UTF-8"): Unit = 
+  val f = java.io.File(fileName)
+  val pw = java.io.PrintWriter(f, enc)
+  try pw.write(text) finally pw.close()
+
+extension (s: String) def isDir: Boolean = java.io.File(s).isDirectory
+
+
+def visitFile(path: String, isModify: Boolean): Unit =
+  print(s"\nprocessing: $path")
+  val lines = loadLines(path)
+  var result = Vector[String]() 
+  for i <- lines.indices do
+    result :+= lines(i)
+    if lines(i).startsWith("scala>") && 
+      lines.lift(i + 1).map(_.startsWith("1 | ")).getOrElse(false) 
+    then
+      result :+= "-- Error:"
+
+  val (msg, isDiff) = 
+    if lines == result then (Console.GREEN + " unmodified" + Console.RESET, false)
+    else (Console.RED + " *** MODIFIED" + Console.RESET, true)
+  println(msg)
+  if isDiff then 
+    println("\nbefore ==========>\n" + lines.mkString("\n"))
+    println("\nafter  ==========>\n" + result.mkString("\n"))
+    if isModify then 
+      println(s"Writing: $path")
+      saveString(result.mkString("\n"), path)
+
+def visitDirRecursively(dirName: String, isModify: Boolean): Unit =
+  println(s"Enter directory: $dirName")
+  list(dirName).foreach( f =>
+    val path = s"$dirName/$f"
+    if path.isDir then visitDirRecursively(path, isModify)
+    else visitFile(path, isModify)
+  )
+
+@main 
+def run_insert_error(isModify: Boolean): Unit = 
+  visitDirRecursively(replTestDir, isModify)
```

Comment:
Nicely done :)

But we don't need to merge this into the repo.

Reply:
Thanks! :smiling_face_with_three_hearts:  

The reason that I committed it to the repo was that I realized that *any* change in REPL output will require similar hacks, to avoid a very tedious manual walkthrough of all tests, so this might needed several times again, as the REPL output, err messages etc will improve over time to get in feature parity with the Scala 2 REPL. 

I have moved the util to this repo now so it is still on the internet if someone needs it again:
https://github.com/lunduniversity/introprog/tree/master/util/repl-output-update-tool

Reply:
Just pushed the delete and the checks are running...

Reply:
Let me know if there is a better place for this little repl-output-update hack to be discoverable to this community (?) and I'll contribute it there.

Reply:
There is `--update-checkfiles` which is passed via `-D` which should (be made to) work for REPL tests.

Reply:
Ok, feel free to use my util code for that if relevant.